src/com/hms_networks/americas/sc/realtimedata/RealTimeDataQueueManager.java

```diff
@@ -148,9 +149,20 @@ public static void dataFetcher(int tagGroup) {
     ArrayList tagGroupArray = (ArrayList) tagManagers.get(tagGroup);
     if (tagGroupArray != null) {
       // for each tag in the tag group, record a new value
-      for (int i = 0; i < tagGroupArray.size(); i++) {
-        TagInfo currentTag = ((TagInfo) ((ArrayList) tagGroupList.get(tagGroup)).get(i));
-        ((RealTimeTagDataPointManager) tagGroupArray.get(i)).recordCurentTagValue(currentTag);
+      for (int iterator = 0; iterator < tagGroupArray.size(); iterator++) {
+        TagInfo currentTag = ((TagInfo) ((ArrayList) tagGroupList.get(tagGroup)).get(iterator));
+        ((RealTimeTagDataPointManager) tagGroupArray.get(iterator))
+            .recordCurentTagValue(currentTag);
+        final int numTagsBeforeSleep = 50;
+        final int sleepMilliseconds = 10;

```

Comment:
How was 10 milliseconds found. Did 1 not work?

Reply:
I just found that this still has problems for OPCUA. with everything being slower I'm not sure if it's the number of tags its waiting, the amount it's sleeping or something else. 1 seemed to work okay on mem tags but when everything is slowed m down from OPCUAI don't think it is enough. @it-hms will probably have to take over this pull request after more testing is done trying different values.

Reply:
changed to 1 for both values for now since they should be better. in a later pr these will need most likely need to be updated